xonsh/tools.py

```diff
@@ -205,6 +216,28 @@ def __eq__(self, other):
         return all(map(operator.eq, self, other))
 
 
+class EnvPathDir(EnvPath):
```

Comment:
@gforsyth Yeah, it makes sense. I'm still have to realize the implementation, which shouldn't be too hard. 

The only concern for me is how to get the tests working. If we have checks like `os.path.isdir()` or `os.path.isfile()` and we parametrize pytest test methods with dummy values, it's bound to fail these checks. We might have to use http://doc.pytest.org/en/latest/tmpdir.html to make it work, but other suggestions are welcome.